framework/src/materials/MaterialPropertyInterface.C

```diff
@@ -225,17 +225,20 @@ MaterialPropertyInterface::addConsumedPropertyName(const MooseObjectName & obj_n
 }
 
 void
-MaterialPropertyInterface::checkMaterialProperty(const std::string & name)
+MaterialPropertyInterface::checkMaterialProperty(const std::string & name, const unsigned int state)
 {
-  // If the material property is boundary restrictable, add to the list of materials to check
-  if (_mi_boundary_restricted)
-    for (const auto & bnd_id : _mi_boundary_ids)
-      _mi_feproblem.storeBoundaryDelayedCheckMatProp(_mi_name, bnd_id, name);
-
-  // The default is to assume block restrictions
-  else
-    for (const auto & blk_ids : _mi_block_ids)
-      _mi_feproblem.storeSubdomainDelayedCheckMatProp(_mi_name, blk_ids, name);
+  if (state == 0)
```

Comment:
Would it make sense to do this `if` at the `getGenericMaterialPropertyByName` level instead?  Otherwise it feels like `checkMaterialProperty` is `checkMaterialPropertyOrMaybeNotJustKidding`